criu/kerndat.c

```diff
@@ -1047,72 +1059,132 @@ int kerndat_init(void)
 	ret = kerndat_try_load_cache();
 	if (ret <= 0)
 		return ret;
+	ret = 0;
 
 	/* kerndat_try_load_cache can leave some trash in kdat */
 	memset(&kdat, 0, sizeof(kdat));
 
 	preload_socket_modules();
 	preload_netfilter_modules();
 
-	ret = check_pagemap();
-	if (!ret)
-		ret = kerndat_get_shmemdev();
-	if (!ret)
-		ret = kerndat_get_dirty_track();
-	if (!ret)
-		ret = init_zero_page_pfn();
-	if (!ret)
-		ret = get_last_cap();
-	if (!ret)
-		ret = kerndat_fdinfo_has_lock();
-	if (!ret)
-		ret = get_task_size();
-	if (!ret)
-		ret = get_ipv6();
-	if (!ret)
-		ret = kerndat_loginuid();
-	if (!ret)
-		ret = kerndat_iptables_has_xtlocks();
-	if (!ret)
-		ret = kerndat_tcp_repair();
-	if (!ret)
-		ret = kerndat_compat_restore();
-	if (!ret)
-		ret = kerndat_tun_netns();
-	if (!ret)
-		ret = kerndat_socket_unix_file();
-	if (!ret)
-		ret = kerndat_nsid();
-	if (!ret)
-		ret = kerndat_link_nsid();
-	if (!ret)
-		ret = kerndat_has_memfd_create();
-	if (!ret)
-		ret = kerndat_detect_stack_guard_gap();
-	if (!ret)
-		ret = kerndat_uffd();
-	if (!ret)
-		ret = kerndat_has_thp_disable();
+	if (check_pagemap()) {
+		pr_err("check_pagemap failed when initializing kerndat.\n");
+		ret = 1;
```

Comment:
ret = -1

nit: usually, we use -1 or another negative value in an error case

Reply:
Ah, thanks for letting me know! That's a good _**one**_ to know :)